cmd/srcd/cmd/sql.go

```diff
@@ -192,81 +162,32 @@ func startGitbaseWithClient(client api.EngineClient) error {
 		return humanizef(err, "could not start gitbase")
 	}
 
-	if err := docker.EnsureInstalled(components.MysqlCli.Image, components.MysqlCli.Version); err != nil {
-		return humanizef(err, "could not install mysql client")
-	}
-
 	return nil
 }
 
-func runMysqlCli(ctx context.Context, query string, opts ...docker.ConfigOption) (*types.HijackedResponse, chan int64, error) {
-	cmd := []string{"mysql", "-h", components.Gitbase.Name}
+func runMysqlCli(ctx context.Context, query string) error {
+	interactive := true
+	cmd := []string{"mysql"}
 	if query != "" {
-		cmd = append(cmd, "-e", query)
-	}
-
-	config := &container.Config{
-		Image: components.MysqlCli.ImageWithVersion(),
-		Cmd:   cmd,
+		cmd = append(cmd, "-t", "-e", query)
+		interactive = false
 	}
-	host := &container.HostConfig{}
-	docker.ApplyOptions(config, host, opts...)
 
-	return docker.Attach(context.Background(), config, host, components.MysqlCli.Name)
-}
+	insResp, err := docker.ExecAndAttach(context.Background(), interactive, components.Gitbase.Name, cmd...)
 
-func attachStdio(resp *types.HijackedResponse) (err error) {
-	inputDone := make(chan error)
-	outputDone := make(chan error)
-
-	in, out, _ := term.StdStreams()
-	// set terminal into raw mode to propagate special characters
-	fd, isTerminal := term.GetFdInfo(in)
-	if isTerminal {
-		var prevState *term.State
-		prevState, err = term.SetRawTerminal(fd)
-		if err != nil {
-			return err
-		}
-		defer func() {
-			err = term.RestoreTerminal(fd, prevState)
-		}()
+	if err != nil {
+		return humanizef(err, "a problem occurred while trying to run mysql client")
 	}
 
-	go func() {
-		_, err := io.Copy(out, resp.Reader)
-		outputDone <- err
-		resp.CloseWrite()
-	}()
-
-	go func() {
-		_, err := io.Copy(resp.Conn, in)
-
-		if err := resp.CloseWrite(); err != nil {
-			log.Debugf("Couldn't send EOF: %s", err)
-		}
-
-		inputDone <- err
-	}()
-
-	select {
-	case err := <-outputDone:
-		return err
-	case err := <-inputDone:
-		if err == nil {
-			// Wait for output to complete streaming.
-			return <-outputDone
-		}
-
-		return err
+	if insResp.Running {
+		return fmt.Errorf("MySQL cli is still running")
 	}
-}
 
-func stopMysqlClient() {
-	err := docker.RemoveContainer(components.MysqlCli.Name)
-	if err != nil {
-		log.Warningf("could not stop mysql client: %v", err)
+	if insResp.ExitCode != 0 {
```

Comment:
not sure if we have tests for it. What happens when you exit with `ctrl-c` or `ctrl-d`? Does it exit correctly with exit code 0?

Reply:
nice catch! Now it exits with 0 on `ctrl-d` but not on `ctrl-c`

Reply:
maybe it's okay to exit with error on `ctrl-c` in this case

Reply:
added a test for `Ctrl-D` [here](https://github.com/src-d/engine/pull/439/commits/4e151e4de579f1d1bb6b0506465875debf117301).